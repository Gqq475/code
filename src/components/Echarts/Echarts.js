import React from 'react'
import classes from './Echarts.scss'
import { TabList, TabPanel, Tabs, Tab } from '@blueprintjs/core'
import EchartsHandlingPath from 'components/EchartsHandlingPath'
import EchartsEquipmentPort from 'components/EchartsEquipmentPort'
import EchartsSTK from 'components/EchartsgSTK'
import EchartsGetFromTo from 'components/EchartsGetFromTo'

type Props = {
  data: Obejct,
  getHandlingPath: Fucntion,
  getEquipmentPort: Fucntion,
  getSTK: Fucntion,
  getFromTo: Fucntion
};

export class Echarts extends React.Component {
  props: Props;

  render () {
    return (
      <div className={classes['Echarts-container']}>
        <Tabs>
          <TabList>
            <Tab>搬运路径共通性分析</Tab>
            <Tab>设备端口共通性分析</Tab>
            <Tab>STK共通性分析</Tab>
            <Tab>起讫点共通性分析</Tab>
          </TabList>
          <TabPanel>
            <EchartsHandlingPath data={this.props.data} getHandlingPath={this.props.getHandlingPath} />
          </TabPanel>
          <TabPanel>
            <EchartsEquipmentPort data={this.props.data} getEquipmentPort={this.props.getEquipmentPort} />
          </TabPanel>
          <TabPanel>
            <EchartsSTK data={this.props.data} getSTK={this.props.getSTK} />
          </TabPanel>
          <TabPanel>
            <EchartsGetFromTo data={this.props.data} getFromTo={this.props.getFromTo} />
          </TabPanel>
        </Tabs>
      </div>
    )
  }
}

export default Echarts
