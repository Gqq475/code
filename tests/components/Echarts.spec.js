import React from 'react'
import Echarts from 'components/Echarts/Echarts'

describe('(Component) Echarts', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <Echarts />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
