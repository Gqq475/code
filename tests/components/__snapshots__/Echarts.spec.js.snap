// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Echarts render normally 1`] = `
<div
  className="Echarts-container"
>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        搬运路径共通性分析
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        设备端口共通性分析
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        STK共通性分析
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        起讫点共通性分析
      </Tab>
    </TabList>
    <TabPanel>
      <EchartsHandlingPath />
    </TabPanel>
    <TabPanel>
      <EchartsEquipmentPort />
    </TabPanel>
    <TabPanel>
      <EchartsgSTK />
    </TabPanel>
    <TabPanel>
      <EchartsGetFromTo />
    </TabPanel>
  </Tabs>
</div>
`;
